source/server/server.cc

```diff
@@ -614,28 +617,29 @@ RunHelper::RunHelper(Instance& instance, const Options& options, Event::Dispatch
       }) {
   // Setup signals.
   if (options.signalHandlingEnabled()) {
-// TODO(Pivotal): Figure out solution to graceful shutdown on Windows. None of these signals exist
-// on Windows.
-#ifndef WIN32
-    sigterm_ = dispatcher.listenForSignal(SIGTERM, [&instance]() {
-      ENVOY_LOG(warn, "caught SIGTERM");
-      instance.shutdown();
-    });
-
-    sigint_ = dispatcher.listenForSignal(SIGINT, [&instance]() {
-      ENVOY_LOG(warn, "caught SIGINT");
-      instance.shutdown();
-    });
-
-    sig_usr_1_ = dispatcher.listenForSignal(SIGUSR1, [&access_log_manager]() {
-      ENVOY_LOG(info, "caught SIGUSR1. Reopening access logs.");
-      access_log_manager.reopen();
-    });
-
-    sig_hup_ = dispatcher.listenForSignal(SIGHUP, []() {
-      ENVOY_LOG(warn, "caught and eating SIGHUP. See documentation for how to hot restart.");
-    });
-#endif
+    if constexpr (platformSupportsSignal(ENVOY_SIGTERM)) {
```

Comment:
Maybe out of scope of this PR and worth doing in a follow up, but it seems like this block could be abstracted out of this file at least, could enable some more detailed test coverage with mocking etc.